internal/controller/clusterdeployment_controller.go

```diff
@@ -505,48 +505,65 @@ func getProjectPolicyRefs(mc *kcm.ClusterDeployment, cred *kcm.Credential) []sve
 	}
 }
 
+func (*ClusterDeploymentReconciler) initServicesConditions(cd *kcm.ClusterDeployment) (changed bool) {
+	for _, typ := range [3]string{kcm.SveltosProfileReadyCondition, kcm.FetchServicesStatusSuccessCondition, kcm.ServicesReferencesValidationCondition} {
+		if apimeta.SetStatusCondition(&cd.Status.Conditions, metav1.Condition{
+			Type:               typ,
+			Status:             metav1.ConditionUnknown,
+			Reason:             kcm.ProgressingReason,
+			ObservedGeneration: cd.Generation,
+		}) {
+			changed = true
+		}
+	}
+
+	return changed
+}
+
+func (*ClusterDeploymentReconciler) setCondition(cd *kcm.ClusterDeployment, typ string, err error) (changed bool) { //nolint:unparam // readability
+	reason, cstatus, msg := kcm.SucceededReason, metav1.ConditionTrue, ""
+	if err != nil {
+		reason, cstatus, msg = kcm.FailedReason, metav1.ConditionFalse, err.Error()
+	}
+
+	return apimeta.SetStatusCondition(&cd.Status.Conditions, metav1.Condition{
+		Type:    typ,
+		Status:  cstatus,
+		Reason:  reason,
+		Message: msg,
+	})
+}
+
 // updateServices reconciles services provided in ClusterDeployment.Spec.ServiceSpec.
 func (r *ClusterDeploymentReconciler) updateServices(ctx context.Context, cd *kcm.ClusterDeployment) (_ ctrl.Result, err error) {
 	l := ctrl.LoggerFrom(ctx)
 	l.Info("Reconciling Services")
 
+	r.initServicesConditions(cd)
+
+	{
+		nsErr := validation.ClusterDeployCrossNamespaceServicesRefs(ctx, cd)
+		tplErr := validation.ServicesHaveValidTemplates(ctx, r.Client, cd.Spec.ServiceSpec.Services, cd.Namespace)
+		merr := errors.Join(nsErr, tplErr)
+		r.setCondition(cd, kcm.ServicesReferencesValidationCondition, merr)
+		if merr != nil {
+			l.Error(merr, "failed to validate services, will not retrigger this error")
```

Comment:
I am thinking what if:

1. User creates CD referencing `myapp` ServiceTemplate, which has `.status.valid=false`.
2. Then afterwards, the `myapp` ServiceTemplate becomes valid.

In this scenario, the CD would not get re-triggered even when the ServiceTemplate is now valid.

Reply:
that is to be done within the #1240 but I could do this as part of this PR, what would you prefer having the context?

Reply:
To be precise, the idea is in the [mgmt controller](https://github.com/k0rdent/kcm/blob/main/internal/controller/management_controller.go#L951-L960) here